driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -290,16 +290,17 @@ private void initClient(final BsonDocument entity, final String id,
             throw new UnsupportedOperationException("Client entity contains unsupported options: " + entity.keySet()
                     + ". Supported options are " + SUPPORTED_CLIENT_ENTITY_OPTIONS);
         }
-        MongoClientSettings.Builder clientSettingsBuilder = getMongoClientSettingsBuilder();
-        if (entity.getBoolean("useMultipleMongoses", BsonBoolean.FALSE).getValue()) {
+        MongoClientSettings.Builder clientSettingsBuilder;
+        if (entity.getBoolean("useMultipleMongoses", BsonBoolean.FALSE).getValue() && (isSharded() || isLoadBalanced())) {
             assumeTrue("Multiple mongos connection string not available for sharded cluster",
                     !isSharded() || getMultiMongosConnectionString() != null);
             assumeTrue("Multiple mongos connection string not available for load-balanced cluster",
                     !isLoadBalanced() || getMultiMongosConnectionString() != null);
-            if (isSharded() || isLoadBalanced()) {
-                clientSettingsBuilder.applyConnectionString(requireNonNull(getMultiMongosConnectionString()));
```

Comment:
This line of code is the one that precipitated the change to the method and the associated change to the Fixture class.  The problem is that the `ClusterSettings.Builder#srvHost` method has the following check:

```
    if (this.hosts != DEFAULT_HOSTS) {
         throw new IllegalArgumentException("Can not set both hosts and srvHost");
     }
```

That exception was being thrown when executing this line of code.  The reason is because on line 293 we are initializing the MongoClientSettings.Builder from the single-mongos URI, and thus `hosts != DEFAULT_HOSTS` by the time we get here.  Moreover, because it's a reference equality check and not an `equals` check, it doesn't even work to first re-set hosts to `singletonList(new ServerAddress())`

I considered a number of solutions, including changing ClusterSettings, but instead opted for this change instead, which I think is actually correct, regardless of whether we change ClusterSettings.